### ISSUE TITLE
Any way to visualize videos or read results without video labeling tool?

### ISSUE THREAD

**Issue by emitoooo** [CMU-CREATE-Lab/deep-smoke-machine#10]
I followed the tutorial; generated "production/2020-11-30" folder and processed the videos. Then created the event/json files.
My question; is there any way to see the model results without installing the video labeling tool?

**Comment by yenchiah** [comment]
Do you mean the classification performance? If yes, they are like f-scores, and there is no need to install the video labeling tool to see that. If you mean that you want to see all the videos that are classified as positive, either you could install the video labeling tool locally, or use python moviepy package (or some other packages).

**Comment by yenchiah** [comment]
More information about moviepy https://github.com/Zulko/moviepy

**Comment by emitoooo** [comment]
Yes, I mean an easy way to see the model results. Let's say I ran xxxx.mp4 video and want to know if it is classified as 1 or 0

**Comment by yenchiah** [comment]
Maybe you can reuse some function in this file https://github.com/CMU-CREATE-Lab/deep-smoke-machine/blob/master/back-end/www/recognize_smoke.py

**Comment by yenchiah** [comment]
I don’t think there is an easy way unfortunately. I remember that I connect PyTorch with tensorboard a long time ago to check the training progress, and tensorboard can show the videos that are classified as 1 or 0

**Comment by emitoooo** [comment]
But what about the evaluation? If I use your pretrained models to test the testset and see the results. I don't check the training progress, I am interested with the evaluation/test performance.

**Comment by yenchiah** [comment]
For the test set I think you will need to modify the code in https://github.com/CMU-CREATE-Lab/deep-smoke-machine/blob/master/back-end/www/i3d_learner.py

For example you can use the add_video function in SummaryWriter in tensorboard
https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_video

**Comment by emitoooo** [comment]
Thanks, I just fixed it. I have another question, can I use pretrained models to test my own videos?
I prepared 36x180x180x3 image-video inputs but couldn't find a way to run the model with custom videos.

**Comment by yenchiah** [comment]
I would suggest using a customized dataset, similar to https://github.com/CMU-CREATE-Lab/deep-smoke-machine/blob/master/back-end/www/smoke_video_dataset.py

Then you can modify the dataloader code in https://github.com/CMU-CREATE-Lab/deep-smoke-machine/blob/master/back-end/www/i3d_learner.py to use your customized dataset
